proxy/run.go

```diff
@@ -184,6 +184,7 @@ func Run(ctx context.Context, args []string) int {
 		Tokens:            cfg.Tokens,
 		Peers:             cfg.Peers,
 		IdempotentGraph:   cfg.IdempotentGraph,
+		ConnectTimeout:    cfg.AstraTimeout,
```

Comment:
This feels like a mixing of concerns.  AstraTimeout is currently used to specify timeouts for metadata and SCB retrieval, both of which are very Astra-specific.  Turning it into a general connect timeout feels like a fairly significant change to it's meaning.

How about if instead we introduce an explicit connect timeout CLI param?  We can then set ConnectTimeout throughout the code base via something like:

--connect-timeout CLI param > --astra-timeout CLI param > default value

Winds up accomplishing very much the same thing while still maintaining the separation of concerns.

Thoughts?

Reply:
A note to future implementors (quite possibly me): there's also an interesting question here about whether we want to make these connect timeouts more granular.  We have potentially distinct timeouts for establishing the control connection (ConnectCluster()) and the connection pools to the various nodes (ConnectSession()).  We may want to enable users to support a longer control connection timeout while enabling more restrictive timeouts for the connection pools.

I'm not :100: convinced this is useful... more something to mull over.